geth/node/manager.go

```diff
@@ -106,6 +103,12 @@ func (m *NodeManager) startNode(config *params.NodeConfig) (<-chan struct{}, err
 			}
 		}()
 
+		// init RPC client for this node
+		m.rpcClient, err = rpc.NewClient(m.node, m.config.UpstreamConfig.URL)
+		if err != nil {
+			log.Error("Init RPC client failed", "error", err)
```

Comment:
@adambabik done. I'm not sure btw, what we should do if RPC client is failed - maybe it's a fatal, and we should panic. But let's keep it this way till further investigation. In case, node is unreachable, underlying client should reconnect automatically without returning errors.